### PR TITLE
Make CompletePresignMultipartUpload required

### LINKED ISSUE #7853: [Bug]: CompletePresignMultipartUpload shoulld be required in completePresignMultipartUpload API call
### What happened?

CompletePresignMultipartUpload is marked as an optional parameter for the CompletePresignMultipartUpload API call but it should be required.

### Expected behavior

_No response_

### lakeFS version

1.25.0

### How lakeFS is installed

_No response_

### Affected clients

_No response_

### Relevant log output

_No response_

### Contact details

_No response_

### PR DESCRIPTION
**Pull request by idanovo**
Closes #7853

### CODE REVIEW

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Wait, that's a breaking change, isn't it?

**Comment by idanovo** [comment]
> Wait, that's a breaking change, isn't it?

@itaiad200  I think it's, but it doesn't make sense to mark the requestBody here as optional, it's a bug.
WDYT we should do about it?

**Comment by arielshaqed** [comment]
I did not know request bodies were optional, that seems _very_ odd.  I would like us to collect some data from The Web about whether required request bodies are a thing.  Does it even change the generated code?  I mean, the difference is that we say the body needs to be at least `{}`, before we say it _also_ needs a `physical_address` field.

AFAICT the existing OpenAPI is fine: if you don't specify a physical address, it won't work.  Since it's in a body, you will need to supply a body.

OTOH, since it doesn't work with a missing body, I agree with @idanovo that this is not a breaking change.  It just makes for weird OpenAPI.  I would appreciate some example of how this improves the developer experience!

**Comment by idanovo** [comment]
> I did not know request bodies were optional, that seems _very_ odd. I would like us to collect some data from The Web about whether required request bodies are a thing. Does it even change the generated code? I mean, the difference is that we say the body needs to be at least `{}`, before we say it _also_ needs a `physical_address` field.
> 
> AFAICT the existing OpenAPI is fine: if you don't specify a physical address, it won't work. Since it's in a body, you will need to supply a body.
> 
> OTOH, since it doesn't work with a missing body, I agree with @idanovo that this is not a breaking change. It just makes for weird OpenAPI. I would appreciate some example of how this improves the developer experience!

@arielshaqed I checked it and after my change, the signature of the function changed from:
`def complete_presign_multipart_upload(self, repository : StrictStr, branch : StrictStr, upload_id : StrictStr, path : Annotated[StrictStr, Field(..., description="relative to the branch")], **complete_presign_multipart_upload : Optional[CompletePresignMultipartUpload] = None**, **kwargs)`
to:
`def complete_presign_multipart_upload(self, repository : StrictStr, branch : StrictStr, upload_id : StrictStr, path : Annotated[StrictStr, Field(..., description="relative to the branch")], **complete_presign_multipart_upload : CompletePresignMultipartUpload**, **kwargs)`

**Review by arielshaqed — CHANGES_REQUESTED** [review]
First of all, thanks!  You have clearly identified a bug in our OpenAPI spec: we identify at runtime developer errors that could be identified at compile time, and at the very least, our Rust typing is overcomplicated

## tl;dr:

This error occurs several times in our OpenAPI spec.  We can change all of them except for 3 without releasing a major version.

We need to decide one of these:
- Fix all internal & experimental endpoints now.
- Fix all endpoints when we release 2.0.

I vote for (2) because it is easier, and because I believe we will have additional breakages before it is time for 2.0 😿 😭 .

So sorry, I am blocking, but _not_ because this particular change is breaking!

## Are there any other cases?
### Are bodies required?

I checked for _other_ occurrences of this.  39 APIs have a request body, 29 of them specify that it is required, 1 explicitly specifies that it is not required, 9 do not specify that it is required:
```sh
❯ yq '.paths.[].[] | select(.requestBody?.required) | .operationId'   ./api/swagger.yml | wc
     29      29     526

❯ yq '.paths.[].[] | select(.requestBody?) | .operationId'   ./api/swagger.yml | wc
     39      39     750

❯ yq '.paths.[].[] | select(.requestBody?.required == false) | .operationId'   ./api/swagger.yml 
```

List of endpoints with a supposedlyoptional request body that contains a required field:
```sh
❯ yq '.paths.[].[] | select(.requestBody? and .requestBody?.required != true) | .operationId'   ./api/swagger.yml 
"login"
"externalPrincipalLogin"
"createUser"
"createGroup"
"createUserExternalPrincipal"
"mergeIntoBranch"
"completePresignMultipartUpload"
"abortPresignMultipartUpload"
"uploadObject"
"prepareGarbageCollectionUncommitted"
```

Of these, these non-internal endpoints require a field in their body:

- login
- createUser
- createGroup

So we have this bug in 3 places where we fixing it will break our compatibility guarantee.

## Is it a breaking change?

> > I did not know request bodies were optional, that seems _very_ odd. I would like us to collect some data from The Web about whether required request bodies are a thing. Does it even change the generated code? I mean, the difference is that we say the body needs to be at least `{}`, before we say it _also_ needs a `physical_address` field.
> > AFAICT the existing OpenAPI is fine: if you don't specify a physical address, it won't work. Since it's in a body, you will need to supply a body.
> > OTOH, since it doesn't work with a missing body, I agree with @idanovo that this is not a breaking change. It just makes for weird OpenAPI. I would appreciate some example of how this improves the developer experience!
> 
> @arielshaqed I checked it and after my change, the signature of the function changed from: `def complete_presign_multipart_upload(self, repository : StrictStr, branch : StrictStr, upload_id : StrictStr, path : Annotated[StrictStr, Field(..., description="relative to the branch")], **complete_presign_multipart_upload : Optional[CompletePresignMultipartUpload] = None**, **kwargs)` to: `def complete_presign_multipart_upload(self, repository : StrictStr, branch : StrictStr, upload_id : StrictStr, path : Annotated[StrictStr, Field(..., description="relative to the branch")], **complete_presign_multipart_upload : CompletePresignMultipartUpload**, **kwargs)`

I am assuming you added double-asterisks around the complete_presign_multipart_upload change.  So I went off and looked at the generated code.

### SDKs

Part of 1.0 was guaranteeing that [minor version changes don't change interfaces](https://docs.lakefs.io/understand/towards-1.0-sdk.html#interface-guarantees).  However we exclude interfaces tagged `experimental` - we get to dodge this for the multipart APIs but not for the auth user APIs.

But let's break it down:
* The Java interface remains unchanged. ✅ 
* The Python interface _also_ remains unchanged, except that code that did not pass complete_multipart_upload now fails typing, whereas before it would fail at runtime. ✅ 
* The Rust interface changes: previously valid code would pass `Some(complete_multipart_upload)`, now it needs to change and pass `complete_multipart_upload`. ❌ 

So I believe that a minor version bump will uphold our SDK guarantee **only** when the API is tagged experimental or internal.

### REST API

The REST API also changes, of course - we now require a body.  But since previously we would error out in this case, I will again argue that the existing cases are bugs, so we are allowed to make the change.

**Comment by idanovo** [comment]
Planned to be fixed as part of version 2.0.0 of lakeFS.
